tasks/changeMultisigOwner.ts

```diff
@@ -0,0 +1,169 @@
+import { task, types } from "hardhat/config";
+import { ethers } from "ethers";
+import * as dotenv from "dotenv";
+import { networkData } from "./utils/networkData";
+
+dotenv.config();
+
+const SafeABI = [
+  "function VERSION() view returns (string)",
+  "function getOwners() view returns (address[])",
+];
+
+const OwnableABI = [
+  "function owner() view returns (address)",
+  "function transferOwnership(address newOwner)",
+];
+
+task("changeMultisigOwner", "Changes the multisig owner")
+  .addParam(
+    "address",
+    "The address of the safe wallet",
+    undefined,
+    types.string,
+    false
+  )
+  .setAction(async (args, hre) => {
+    const network = hre.network.name;
+    const typedArgs = args as { address: string };
+    const address = typedArgs.address;
+    console.log(`Changing multisig owner to: ${address} - ${network}`);
+
+    if (!network || !address) {
+      console.error("Usage: npm run multisigOwner <network> <address>");
+      return;
+    }
+
+    const currentNetworkData = networkData[network as keyof typeof networkData];
+
+    const { owners, lbcProxyAddress, lbcProxyAdminAddress } =
+      currentNetworkData;
+
+    const etherProvider = new ethers.JsonRpcProvider(
+      currentNetworkData.providerRpc
+    );
+
+    const isSafe = await isSafeAddress(etherProvider, address);
+    if (!isSafe) {
+      console.error(
+        "Exiting... Provided Safe address is not a valid Safe contract."
+      );
+      return;
+    }
+
+    const safeOwners = await getOwners(etherProvider, address);
+    if (!validateOwners(safeOwners, owners)) {
+      console.error(
+        "Exiting... Safe owners do not match expected configuration."
+      );
+      return;
+    }
+
+    const signer = ethers.Wallet.fromPhrase(
+      currentNetworkData.mnemonic!,
+      etherProvider
+    );
+
+    console.log("Starting ownership transfer process...");
+
+    await transferOwnership(signer, lbcProxyAddress, address);
+    await transferOwnership(signer, lbcProxyAdminAddress, address);
+
+    console.log("Ownership transfer process complete!");
+  });
+
+async function isSafeAddress(
+  provider: ethers.JsonRpcProvider,
+  address: string
+): Promise<boolean> {
+  try {
+    const code = await provider.getCode(address);
+    if (code === "0x") {
+      console.log(`${address} is not a smart contract`);
+      return false;
+    }
+    const contract = new ethers.Contract(address, SafeABI, provider);
+
+    const version = (await contract.VERSION()) as string;
+    console.log(`Address ${address} is a Safe contract! Version: ${version}`);
+
+    return true;
+  } catch (error) {
+    console.error(`Address ${address} Is not a Safe address.`, error);
+    return false;
+  }
+}
+
+async function getOwners(
```

Comment:
based in my other comments, I recommend to remove this function

Reply:
solved